engine/extended_provider/extended_providers_publisher.go

```diff
@@ -0,0 +1,155 @@
+package extended_provider
```

Comment:
What is the rationale for separating the package? Would be nice if we don't have to do this and the functionality simply extends the existing API?

Reply:
My original idea was to add this API into the main interface, however if you think about it - there is nothing fundamentally new that has been added. I.e. the core engine functionality and the APIs that are exposed from it already are sufficient for the extended providers usecase too. 

Hence instead of modifying the existing APIs that would require at least 2 new parameters (override and ExtendedProviders) I decided to go with a layer on top of the existing engine. I also envision that the extended providers APIs are going to be used way less frequent comparing to the main `Publish` and `NotifyPut`. So modifying those to satisfy extended providers usecase was not an option either.